### PR TITLE
geolocation viewport for pie

### LINKED ISSUE #150: Backend: respect geolocation viewport for pieplot
related to veupathdb/edadataservice#166 . plot.data needs to be updated to reflect the new api and to properly determine completeCases* counts

### PR DESCRIPTION
**Pull request by d-callan**
resolves #150

### CODE REVIEW

File: R/class-plotdata.R [hunk 1]
```diff
@@ -86,6 +101,7 @@ newPlotdata <- function(.dt = data.table(),
     veupathUtils::logWithTime(paste('Replaced NA with 0 in the following columns: ', impute0cols), verbose)
   }
 
+  #lat and lon must be used w a geohash, so they dont need to be part of completeCases*
```
Comment (asizemore) [inline R/class-plotdata.R:104]: i dont follow this part - why does being used with a geohash make a difference here?

File: R/utils.R [hunk 2]
```diff
@@ -229,6 +242,30 @@ adjustToViewport <- function(x, viewport) {
...
 
+#' Filter Data Table to Geolocation Viewport
+#' 
+#'
+#' This function will filter a data.table to the specified 
+#' latitude and longitude values.
+#' @param .dt data.table with columns for latitude and longitude
+#' @param latitude Character vector specifying name of latitude column
+#' @param longitude Character vector specifying name of longitude column
+#' @param viewport List of lists specifying the geolocation viewport
```
Comment (asizemore) [inline R/utils.R:253]: also may be helpful to specify list names that are expected, like xMin and left/right.

File: R/utils.R [hunk 3]
```diff
@@ -229,6 +242,30 @@ adjustToViewport <- function(x, viewport) {
...
+#' @return data.table of filtered values 
+#' @export
+filterToGeolocationViewport <- function(.dt, latitude, longitude, viewport) {
+  if (is.null(viewport)) { return(.dt) }  
+
+  .dt <- .dt[.dt[[latitude]] >= viewport$latitude$xMin & .dt[[latitude]] <= viewport$latitude$xMax]
+  if (viewport$longitude$left < viewport$longitude$right) {
```
Comment (asizemore) [inline R/utils.R:260]: Note: here we keep data on the viewport boundaries (with <= and => ) but in the [data service](https://github.com/VEuPathDB/EdaDataService/blob/f1883ce9e71c5bbdaedeb5e9e42007424fa298cf/src/main/java/org/veupathdb/service/eda/ds/plugin/pass/MapPlugin.java#L110-L115) we do not

**Review by asizemore — COMMENTED** [review]
After reading this PR i'm also in favor of making map markers their own subclass :)

File: R/class-plotdata.R (hunk 1 above)
Comment (d-callan) [reply R/class-plotdata.R:104]: my understanding from bob is that if there is a geohash, there is latitude and longitude.

File: R/utils.R (hunk 3 above)
Comment (d-callan) [reply R/utils.R:260]: the logic is reversed bc one is inclusive and the other exclusive. result should be the same.

File: R/class-plotdata.R (hunk 1 above)
Comment (d-callan) [reply R/class-plotdata.R:104]: tbh this assumption in combination w the fact that any time we use geohash we seem to also need lat and lon make me think the three might be a candidate for a class when we switch to s4.

**Review by asizemore — APPROVED** [review]
looks good to me!

**Pull request merged**